core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1048,7 +1065,7 @@ private[spark] class BlockManager(
           try {
             replicate(blockId, bytesToReplicate, level, remoteClassTag)
           } finally {
-            bytesToReplicate.dispose()
+            bytesToReplicate.unmap()
```

Comment:
yea we should separate `unmap` and `dispose`. but instead of using a hack in https://github.com/apache/spark/pull/16499/files#diff-21027f5c826cd378daaae5f7c3eea2b5R240, shall we use a just a flag `needUnmap` in `ChunkedByteBuffer `?